src/current/v23.2/read-committed.md

```diff
@@ -0,0 +1,953 @@
+~~~
+
+~~~
+     day     | doctor_id | on_call
+-------------+-----------+----------
+  2023-12-05 |         1 |    t
+  2023-12-05 |         2 |    f
+~~~
+  </div>
+
+  <div class="grid-item">
+In Session 1, commit the transaction:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+COMMIT;
+~~~
+
+[By design under `READ COMMITTED` isolation]({% link {{ page.version.version }}/architecture/transaction-layer.md %}#write-skew-tolerance), CockroachDB allows the transaction to commit even though its previous read (the `SELECT` query) has changed due to the concurrent transaction in Session 2.
+  </div>
+
+  <div class="grid-item">
+In Session 2, read the rows for `2023-12-05` again:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SELECT * FROM schedules
+  WHERE day = '2023-12-05';
+~~~
+
+~~~
+     day     | doctor_id | on_call
+-------------+-----------+----------
+  2023-12-05 |         1 |    f
+  2023-12-05 |         2 |    f
+~~~
+
+The result has changed because Session 1 committed earlier and updated the `on_call` value for doctor 1, thus changing the read result for the transaction in Session 2.
+
+If the transaction in Session 2 commits and updates the `on_call` value for Betty, this will create a [write skew anomaly](#write-skew-anomaly). The result would be that neither Abe nor Betty is scheduled to be on call on `2023-12-05`.
+
+Instead, the transaction should rollback so that the write skew anomaly does not commit:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+ROLLBACK;
+~~~
+  </div>
+</div>
+
+### Reserve rows for updates using exclusive locks
+
+Before proceeding, reset the [example scenario](#before-you-begin):
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+UPDATE schedules SET on_call = true WHERE on_call = false;
+~~~
+
+Confirm that at least one doctor is on call each day of the week:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SELECT day, count(*) AS on_call FROM schedules
+  WHERE on_call = true
+  GROUP BY day
+  ORDER BY day;  
+~~~
+
+~~~
+     day     | on_call
+-------------+----------
+  2023-12-01 |       2
+  2023-12-02 |       2
+  2023-12-03 |       2
+  2023-12-04 |       2
+  2023-12-05 |       2
+  2023-12-06 |       2
+  2023-12-07 |       2
+~~~
+
+  <div class="grid-container">
+  <h5 style="text-align:center">Session 1</h5><h5 style="text-align:center">Session 2</h5>
+  <div class="grid-item">
+Doctor 1, Abe, starts to request leave for `2023-12-05` using the hospital's schedule management application.
+
+Start a transaction:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED;
+~~~
+
+Check to make sure that another doctor is on call for `2023-12-05`. Use [`FOR UPDATE`]({% link {{ page.version.version }}/select-for-update.md %}) to lock the rows so that only the current transaction can update them:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SELECT * FROM schedules
+  WHERE day = '2023-12-05'
+  FOR UPDATE;
```

Comment:
To avoid deadlock, both SFU should have `ORDER BY doctor_id FOR UPDATE` (and maybe this is worth pointing out!).